lib/ansible/module_utils/aws/core.py

```diff
@@ -120,14 +128,38 @@ def __init__(self, **kwargs):
         self._diff = self._module._diff
         self._name = self._module._name
 
+        self._botocore_endpoint_log_stream = StringIO()
+        self.logger = None
+        if self.params['debug_botocore_endpoint_logs']:
+            self.logger = logging.getLogger('botocore.endpoint')
+            self.logger.setLevel(logging.DEBUG)
+            self.logger.addHandler(logging.StreamHandler(self._botocore_endpoint_log_stream))
+
     @property
     def params(self):
         return self._module.params
 
+    def _get_resource_action_list(self):
+        actions = []
+        for ln in self._botocore_endpoint_log_stream.getvalue().split('\n'):
+            ln = ln.strip()
+            if not ln:
+                continue
+            found_operational_request = re.search(r"OperationModel\(name=.*?\)", ln)
+            if found_operational_request:
+                operation_request = found_operational_request.group(0)[20:-1]
+                resource = re.search("https://.*?\.", ln).group(0)[8:-1]
+                actions.append("{0}:{1}".format(resource, operation_request))
```

Comment:
It would be nice if this was a list of (action, resource_arn) pairs, but I'm still trying to find any ARN logging

Reply:
@mattclay Searching to no avail, the closest I can get to generating resource restrictions is by scraping the AWS documentation. The targeted resources are not logged and I can't find a way to construct them from anything in botocore.

Here's a script that takes the output from this callback and creates a policy with the documented resource restrictions: https://gist.github.com/s-hertel/00ffb9285c417c5955eab63a042a8c32.